components/automate-ui/src/app/pages/automate-settings/automate-settings.component.ts

```diff
@@ -102,63 +197,102 @@ export class AutomateSettingsComponent implements OnInit {
           this.jobSchedulerStatus = automateSettingsSelector.jobSchedulerStatus;
           this.telemetryService.track('lifecycleConfiguration', this.jobSchedulerStatus);
           this.updateForm(this.jobSchedulerStatus);
-          this.onChanges();
         }
       });
-  }
 
-  // Has the form changed?
-  public noChanges() {
-    return !this.formChanged;
   }
 
-  // patchUnitValue is the workaround for the chef-select molecule since it is not
-  // an input annotation we need to patch the value inside the FormGroup
-  public patchUnitValue(form, event) {
-    form.controls['unit'].setValue(event.target.value);
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
   }
 
-  // isUnitValue is the workaround for the chef-select molecule since it is not
-  // a select field we need to check value to use inside FormGroup
-  public isUnitValue(form, unit: string) {
-    return form.controls['unit'].value === unit;
+  // This prevents a user from being allowed to enter negative numbers
+  // or other actions that we dont want to allow
+  public preventNegatives(key: string) {
+    const allowedKeys = [
+      '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+      'Backspace', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown', 'Tab'
+    ];
+    if ( !allowedKeys.includes(key) ) {
+      return false;
+    }
   }
 
-  // patchDisableValue is the workaround for the chef-checkbox molecule since it is not
-  // an input annotation we need to patch the value inside the FormGroup
-  public patchDisableValue(form, checked: boolean) {
-    form.controls['disable'].setValue(checked);
+  public toggleInput(form, checked: boolean): void {
```

Comment:
It would be more clear to rename this to either `setInput` or `setState`. You are not toggling the input; you are setting it to the given value. (The giveaway is that a true toggle method would not need a boolean parameter because it is just toggling.)

Reply:
I renamed it to 'handleFormActivation' because in essence it is either activating, or deactivating the form based on the click